sky/serve/serve_utils.py

```diff
@@ -356,20 +357,32 @@ def wait_service_initialization(service_name: str, job_id: int) -> str:
     cnt = 0
     while True:
         record = serve_state.get_service_from_name(service_name)
-        if record is None:
-            continue
-        if job_id != record['controller_job_id']:
-            return common_utils.encode_payload(None)
-        lb_port = record['load_balancer_port']
-        if lb_port is not None:
-            return common_utils.encode_payload(lb_port)
+        if record is not None:
+            if job_id != record['controller_job_id']:
```

Comment:
This is a bit hacky. It should be fine for now, but let's have a comment here for why the inequality means the name conflict. : )

Reply:
I've added `See sky/execution.py::serve_up for more details.` in the function docstr. Do you think we still need a comment here?